src/test_solver.jl

```diff
@@ -0,0 +1,63 @@
+import Base.Test: @test, @test_approx_eq_eps
+
+const testset = [
+                 Dict(:ivp      => ExplicitODE(0.0,
+                                               [0.0],
+                                               (t,y,dy)->dy[1]=6.0,
+                                               J! =(t,y,dy)->dy[1]=0.0),
+                      :sol      => t->[6t],
+                      :isscalar => true,
+                      :name     => "y'=6",
+                      :options  => Dict(:initstep => 0.1,
+                                        :tstop => 1.0)
+                      )
+                 ]
+
+
+function test_integrator{I<:AbstractIntegrator}(integrator::Type{I};
+                                                params = nothing,
+                                                eqntypes = [ExplicitODE])
+
+    const tol = 0.02
+
+    # filter the tests
+    tests = filter(x->in(typeof(x[:ivp]),eqntypes),testset)
+
+    for test in testset
```

Comment:
Would it make sense to make each sub-test a function?  That would make documenting the test easier and would allow adding more tests without making this function huge?